### PR TITLE
Document options to avoid Docker Hub rate limit

### PR DESCRIPTION
**Pull request by chanseokoh**
I realized this is important to document.

I'll also expand the FAQ to cover the options not explained in this PR.

### CODE REVIEW

File: docs/default_base_image.md [hunk 1]
```diff
@@ -8,6 +8,32 @@ For WAR projects, the default is the official [`jetty`](https://hub.docker.com/_
...
+        "errors": [
+          {
+            "code": "TOOMANYREQUESTS",
+            "message": "You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit"
+          }
+        ]
+      }
+```
+Note that, even after Jib fully cached a base image, Jib still connects to Docker Hub to check the imaage every time it runs (unless you pin to a specific base image using a SHA digest). This is to check if the cached base image is up-to-date.
+
```
Comment (mpeddada1) [inline docs/default_base_image.md:27]:
```suggestion
Note that, even after Jib fully cached a base image, Jib still connects to Docker Hub to check the image every time it runs (unless you pin to a specific base image using a SHA digest). This is to check if the cached base image is up-to-date.

```

File: docs/default_base_image.md [hunk 2]
```diff
@@ -8,6 +8,32 @@ For WAR projects, the default is the official [`jetty`](https://hub.docker.com/_
...
+Some options:
+* Configure a registry mirror.
```
Comment (mpeddada1) [inline docs/default_base_image.md:29]: let's add a link to the faq here

File: docs/default_base_image.md [hunk 3]
```diff
@@ -8,6 +8,32 @@ For WAR projects, the default is the official [`jetty`](https://hub.docker.com/_
...
+* Prevent Jib from accessing Docker Hub (after Jib cached a base image locally).
+  * Pin to a specific base image using a SHA digest (for example, `jib.from.image=adoptopenjdk:11-jre@sha256:...`).
+  * Do offline building.
```
Comment (mpeddada1) [inline docs/default_base_image.md:32]:
```suggestion
  * Do offline building using `--offline`.
```

File: docs/default_base_image.md [hunk 4]
```diff
@@ -8,6 +8,32 @@ For WAR projects, the default is the official [`jetty`](https://hub.docker.com/_
...
+* Retry with increasing backoffs.
+
+See this [FAQ](https://github.com/GoogleContainerTools/jib/blob/master/docs/faq.md#i-am-hitting-docker-hub-rate-limits-how-can-i-configure-registry-mirrors) for more details about the options.
```
Comment (mpeddada1) [inline docs/default_base_image.md:35]: This FAQ only talks about registry mirrors. We may not needed if we mention it above?

**Comment by chanseokoh** [comment]
I will expand the FAQ entry (as I said in the PR description). I think it's worth covering all the options in both places but don't have to duplicate the contents. We can briefly mention the options here and cover others in the FAQ.

File: docs/default_base_image.md (hunk 2 above)
Comment (chanseokoh) [reply docs/default_base_image.md:29]: See https://github.com/GoogleContainerTools/jib/pull/3198#issuecomment-816715782

File: docs/default_base_image.md (hunk 3 above)
Comment (chanseokoh) [reply docs/default_base_image.md:32]: I intentionally don't want to mention `--offline` without explaining more details, as the user can easily run into a problem. That is, see https://github.com/GoogleContainerTools/jib/pull/3198#issuecomment-816715782

File: docs/default_base_image.md (hunk 4 above)
Comment (mpeddada1) [reply docs/default_base_image.md:35]: Oops my bad, I didn't see the comment above.

File: docs/default_base_image.md (hunk 3 above)
Comment (mpeddada1) [reply docs/default_base_image.md:32]: Okay, that's a valid point.

**Review by mpeddada1 — APPROVED** [review]

**Pull request merged**